### PR TITLE
Remove download url and sha ENV instructions in 2.1 Dockerfiles

### LINKED ISSUE #499: Reduce the number of layers in the Alpine runtime images
```
FROM microsoft/dotnet-nightly:2.1-runtime-deps-alpine

# Install .NET Core
ENV DOTNET_VERSION 2.1.0-preview1-25919-02
ENV DOTNET_DOWNLOAD_URL https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-alpine.3.6-x64.tar.gz
ENV DOTNET_DOWNLOAD_SHA 7a8b081c890226ba7220c654dc833752f837171c2fe449367d24c2980db3a809012478a67423fd46264e5d2f0389a6890077f8dd25a1fc4aeecb25deb84d72a4

RUN apk add --no-cache --virtual .build-deps \
        openssl \
    && wget -O dotnet.tar.gz $DOTNET_DOWNLOAD_URL \
    && echo "$DOTNET_DOWNLOAD_SHA  dotnet.tar.gz" | sha512sum -c - \
    && mkdir -p /usr/share/dotnet \
    && tar -C /usr/share/dotnet -xzf dotnet.tar.gz \
    && ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet \
    && rm dotnet.tar.gz \
    && apk del .build-deps
```

The three ENV instructions are not necessary.  The various official Docker images are not consistent on this pattern - some have it and others don't.  Since the purpose of supporting Alpine is to have an efficient and slime image, IMO we should remove the URL and SHA layers.  Their value is limited.  I believe there is value in preserving the DOTNET_VERSION as this easily identifies the version of the primary component.  Seeing this called out in the Dockerfile and when viewing Docker image layers is useful.

If this change is made, it should be used across all new Dockerfiles going forward in order to have consistency.  Additionally, the update-dependency tool will need to be updated accordingly.

### PR DESCRIPTION
**Pull request by MichaelSimons**
Fixes #499

### CODE REVIEW

File: 2.1/runtime/alpine/amd64/Dockerfile [hunk 1]
```diff
@@ -2,13 +2,12 @@ FROM microsoft/dotnet-nightly:2.1-runtime-deps-alpine
...
 # Install .NET Core
 ENV DOTNET_VERSION 2.1.0-preview1-25919-02
-ENV DOTNET_DOWNLOAD_URL https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-alpine.3.6-x64.tar.gz
-ENV DOTNET_DOWNLOAD_SHA 7a8b081c890226ba7220c654dc833752f837171c2fe449367d24c2980db3a809012478a67423fd46264e5d2f0389a6890077f8dd25a1fc4aeecb25deb84d72a4
 
 RUN apk add --no-cache --virtual .build-deps \
         openssl \
-    && wget -O dotnet.tar.gz $DOTNET_DOWNLOAD_URL \
-    && echo "$DOTNET_DOWNLOAD_SHA  dotnet.tar.gz" | sha512sum -c - \
+    && wget -O dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-alpine.3.6-x64.tar.gz \
```
Comment (ravimeda) [inline 2.1/runtime/alpine/amd64/Dockerfile]: `https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-alpine.3.6-x64.tar.gz` - safer to put this in quotes.

File: 2.1/runtime/stretch/amd64/Dockerfile [hunk 2]
```diff
@@ -7,11 +7,10 @@ RUN apt-get update \
 
 # Install .NET Core
 ENV DOTNET_VERSION 2.1.0-preview1-25919-02
-ENV DOTNET_DOWNLOAD_URL https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-linux-x64.tar.gz
-ENV DOTNET_DOWNLOAD_SHA E215DFD387C569707805BB693B7B006B124A2A13AA6F2C2CD9DCC8C1B34ED221A09302BC6D337276BB6CD6D13CC7A2675DE9B7BCF956D69C373D9B177446E14B
 
-RUN curl -SL $DOTNET_DOWNLOAD_URL --output dotnet.tar.gz \
-    && echo "$DOTNET_DOWNLOAD_SHA dotnet.tar.gz" | sha512sum -c - \
+RUN curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Runtime/$DOTNET_VERSION/dotnet-runtime-$DOTNET_VERSION-linux-x64.tar.gz \
```
Comment (ravimeda) [inline 2.1/runtime/stretch/amd64/Dockerfile]: Not related to your change. But a clarifying question -  How do you determine whether to use `curl` or `wget`?

File: 2.1/runtime/alpine/amd64/Dockerfile (hunk 1 above)
Comment (MichaelSimons) [reply 2.1/runtime/alpine/amd64/Dockerfile]: The url doesn't have spaces - leaving as is.

File: 2.1/runtime/stretch/amd64/Dockerfile (hunk 2 above)
Comment (MichaelSimons) [reply 2.1/runtime/stretch/amd64/Dockerfile]: `curl` has been the standard used in the .NET Dockerfiles however it is not available in the Alpine base image therefore `wget` is used for Alpine

**Pull request merged**